### PR TITLE
feat: parse region from ARN

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Parse AWS Region if the secret ID provided to the Secrets Manager connection plugin is an ARN string, so that secretsManagerRegion is no longer a required parameter for ARNs.

Addresses #391  

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by congoamz — APPROVED** [review]
There are a couple qodana warnings, not sure if they're relevant, otherwise looks good to me

**Review by joyc-bq — APPROVED** [review]
lgtm

File: docs/using-the-jdbc-driver/using-plugins/UsingTheAwsSecretsManagerPlugin.md [hunk 1]
```diff
@@ -8,15 +8,15 @@ The AWS Advanced JDBC Driver supports usage of database credentials stored as se
...
 
-| Parameter                | Value  | Required | Description                                             | Example     | Default Value |
-|--------------------------|:------:|:--------:|:--------------------------------------------------------|:------------|---------------|
-| `secretsManagerSecretId` | String |   Yes    | Set this value to be the secret name or the secret ARN. | `secretId`  | `null`        |
-| `secretsManagerRegion`   | String |   Yes    | Set this value to be the region your secret is in.      | `us-east-2` | `us-east-1`   |
+| Parameter                | Value  |                        Required                         | Description                                             | Example     | Default Value |
+|--------------------------|:------:|:-------------------------------------------------------:|:--------------------------------------------------------|:------------|---------------|
+| `secretsManagerSecretId` | String |                           Yes                           | Set this value to be the secret name or the secret ARN. | `secretId`  | `null`        |
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/using-plugins/UsingTheAwsSecretsManagerPlugin.md:17]: Can we provide an example of AWS ARN?

File: wrapper/src/main/java/software/amazon/jdbc/plugin/AwsSecretsManagerConnectionPlugin.java [hunk 2]
```diff
@@ -109,12 +114,18 @@ public AwsSecretsManagerConnectionPlugin(PluginService pluginService, Properties
...
-    final String regionString = REGION_PROPERTY.getString(props);
-    if (StringUtils.isNullOrEmpty(regionString)) {
-      throw new RuntimeException(
-          Messages.get(
-              "AwsSecretsManagerConnectionPlugin.missingRequiredConfigParameter",
-              new Object[] {REGION_PROPERTY.name}));
+    final Matcher matcher = SECRETS_ARN_PATTERN.matcher(secretId);
+    String regionString;
+    if (matcher.matches()) {
+      regionString = matcher.group("region");
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/AwsSecretsManagerConnectionPlugin.java:120]: User should be able to override region from ARN.

File: wrapper/src/test/java/software/amazon/jdbc/plugin/AwsSecretsManagerConnectionPluginTest.java [hunk 3]
```diff
@@ -356,10 +357,36 @@ public void testConnectWithWrappedPostgreSQLException() throws SQLException {
     assertEquals(TEST_PASSWORD, TEST_PROPS.get(PropertyDefinition.PASSWORD.name));
   }
 
+  @ParameterizedTest
+  @MethodSource("arnArguments")
+  public void testConnectViaARN(final String arn, final Region region) throws SQLException {
```
Comment (sergiyvamz) [inline wrapper/src/test/java/software/amazon/jdbc/plugin/AwsSecretsManagerConnectionPluginTest.java:362]: I think we need another test to check optional/required parameters (secretId and region)

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**